ci/local/build.bash

```diff
@@ -0,0 +1,183 @@
+#! /usr/bin/env bash
+
+# Copyright (c) 2018-2020 NVIDIA Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Released under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+
+################################################################################
+# Thrust and CUB local containerized build script
+################################################################################
+
+function usage {
+  echo "Usage: ${0} [flags...] [cmake-targets...]"
+  echo
+  echo "Build and test your local repository using a gpuCI Docker image."
+  echo "If CMake targets are specified, only those targets are built and tested."
```

Comment:
FYI, this will only work with non-meta targets, since CTest doesn't know about meta targets.

This means that targets for specific tests (`thrust.cpp.cuda.cpp17.test.scan`) will work, but entire configs (`thrust.cpp.cuda.cpp17.all`) will not. If you want to latter to work let me know and we can figure something out.

Reply:
I'd like the latter to work. Should I file an issue?

Reply:
Please do.

To get partway there, I think we can just drop the `.all` from the metatargets. Then `ninja thrust.cpp.cuda.cpp17` and `ctest -R thrust.cpp.cuda.cpp17` would both work as expected.

There's another issue -- doing multiple configs at once will be tricker because `ctest -R` behaves kinda weird. We'd have to join the configs into a regex to get it to actually run all tests from those configs:

```bash
# Works easily on ninja:
ninja thrust.cpp.cuda.cpp11 thrust.cpp.cuda.cpp14 thrust.cpp.cuda.cpp17
# Doesn't work, -R only accepts a single argument:
ctest -R thrust.cpp.cuda.cpp11 thrust.cpp.cuda.cpp14 thrust.cpp.cuda.cpp17
# Doesn't work, only one -R can be specified, the final overwrites the others and only cpp17 tests run
ctest -R thrust.cpp.cuda.cpp11 -R thrust.cpp.cuda.cpp14 -R thrust.cpp.cuda.cpp17
# Works, but it's kinda horrible because it's really a regex:
ctest -R "thrust.cpp.cuda.cpp11|thrust.cpp.cuda.cpp14|thrust.cpp.cuda.cpp17"
# What we are really trying to say:
ctest -R "^thrust\.cpp\.cuda\.cpp11|^thrust\.cpp\.cuda\.cpp14|^thrust\.cpp\.cuda\.cpp17"
```

So some bash string manip will be needed to get to something useable for multiple targets.

Reply:
Actually, it looks like multiple -R regexes can be combined when -U is passed. So this will also work (though again, being sloppy with regexes):

```
ctest -U -R thrust.cpp.cuda.cpp11 -R thrust.cpp.cuda.cpp14 -R thrust.cpp.cuda.cpp17
```

TIL the "-U" is for union, and there's also a "-I" for intersection.

Not sure if this buys us anything since we'd still have to process the string, but it's definitely cleaner than generating a single mega-regex.

Reply:
I've fixed this for now, it will work with either no target or specific targets.